### PR TITLE
feat: add JSDoc style comment support to Typescript generator

### PR DESCRIPTION
**Pull request by gabormagyar**
Might need tests; tests for comment generation are already included just not for them being included. Don't think it needs docs as the feature is automatic and self explanatory.

Make sure to filter whitespace changes when viewing diff.

### CODE REVIEW

File: src/generators/typescript/TypeScriptRenderer.ts [hunk 1]
```diff
@@ -156,17 +156,24 @@ ${renderedLines}
 
   renderProperty(propertyName: string, property: CommonModel, type: PropertyType = PropertyType.property): string {
     const formattedPropertyName = this.nameProperty(propertyName, property);
+
+    const desc = property.getFromOriginalInput('description').trim();
+    const examples = property.getFromOriginalInput('examples');
+    const formattedExamples = `@example ${examples?.join ? examples.join(', ') : examples}`
+    const doc = this.renderComments(`${desc || ''}\n${examples ? formattedExamples : ''}`.trim());
+    const formattedDoc = desc || examples ? `${doc}\n` : '';
```
Comment (jonaslagoni) [inline src/generators/typescript/TypeScriptRenderer.ts:164]: Remember, any of these properties fetched from the original input might not exist at all, so to use them you need extensive checks ensuring that you find anything 🙂

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Small specific comment, also remember to add tests to ensure it works as intended 👍 

It seems you might have auto-formatting enabled in your development environment making unnecessary changes that make it harder to review, do you mind disabling it and re-commit the intended changes? 🙂

**Comment by gabormagyar** [comment]
Auto formatting removed

**Comment by jonaslagoni** [comment]
@gabormagyar sorry I did not caught this the first time, this has to be added as a preset instead, similar to what is being done here: https://github.com/asyncapi/modelina/pull/649 and what we have for Java https://github.com/asyncapi/modelina/blob/master/docs/languages/Java.md#include-javadoc-for-properties already.

We need to remember, that it's up to the users how the output model should sticked together 🙂

**Comment by gabormagyar** [comment]
@jonaslagoni Please re-review

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Looks good 👍 

We are only missing an example and updating the documentation: https://github.com/asyncapi/modelina/blob/master/docs/languages/JavaScript.md

Otherwise there are no easy way for others to know about it 🙂

**Review by Samridhi-98 — CHANGES_REQUESTED** [review]
@gabormagyar  Please add `typscript-generate-comment` [here](https://github.com/asyncapi/modelina/tree/master/examples#readme). 😊

**Review by jonaslagoni — COMMENTED** [review]
That was quick ⏩!

This looks good to me 👍

**Comment by Samridhi-98** [comment]
> That was quick ⏩!
> 
> This looks good to me 👍
> 
> @Samridhi-98 do you want to make a review as well?

I checked rest looks good to me 🙂

**Review by Samridhi-98 — DISMISSED** [review]
🎉

**Comment by jonaslagoni** [comment]
Got some linting problems 🙂

**Review by Samridhi-98 — APPROVED** [review]

**Review by jonaslagoni — APPROVED** [review]

**Comment by Samridhi-98** [comment]
@all-contributors please add @gabormagyar  for code, test, docs, example

**Pull request merged**